src/RadixDlt.NetworkGateway.PostgresIntegration/Models/EntityResourceAggregateHistory.cs

```diff
@@ -84,12 +86,41 @@ internal class EntityResourceAggregateHistory
     [Column("fungible_resource_entity_ids")]
     public List<long> FungibleResourceEntityIds { get; set; }
 
-    [Column("fungible_resource_last_update_state_versions")]
-    public List<long> FungibleResourceLastUpdateStateVersions { get; set; }
+    [Column("fungible_resource_significant_update_state_versions")]
+    public List<long> FungibleResourceSignificantUpdateStateVersions { get; set; }
 
     [Column("non_fungible_resource_entity_ids")]
     public List<long> NonFungibleResourceEntityIds { get; set; }
 
-    [Column("non_fungible_resource_last_update_state_versions")]
-    public List<long> NonFungibleResourceLastUpdateStateVersions { get; set; }
+    [Column("non_fungible_resource_significant_update_state_versions")]
+    public List<long> NonFungibleResourceSignificantUpdateStateVersions { get; set; }
+
+    public void MarkAsCloneOfExistingRecord()
+    {
+        _fungibleResourceEntityIdsOrg = FungibleResourceEntityIds.ToArray();
+        _nonFungibleResourceEntityIdsOrg = NonFungibleResourceEntityIds.ToArray();
+    }
+
+    public bool ShouldBePersisted()
```

Comment:
that piece of code would be easily testable by unit tests, maybe we should add some?

Reply:
Refactored EntityResourceAggregateHistory quite heavily.